homeassistant/components/shiftr.py

```diff
@@ -0,0 +1,80 @@
+"""
+Support for Shiftr.io.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/shiftr/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    CONF_PASSWORD, CONF_USERNAME, EVENT_STATE_CHANGED,
+    EVENT_HOMEASSISTANT_STOP)
+from homeassistant.helpers import state as state_helper
+
+REQUIREMENTS = ['paho-mqtt==1.2.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'shiftr'
+
+SHIFTR_BROKER = 'broker.shiftr.io'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Initialize the Shiftr.io MQTT consumer."""
+    import paho.mqtt.client as mqtt
+    conf = config[DOMAIN]
+    username = conf.get(CONF_USERNAME)
+    password = conf.get(CONF_PASSWORD)
+
+    client_id = 'HomeAssistant'
+    port = 1883
+    keepalive = 600
+
+    mqttc = mqtt.Client(client_id, protocol=mqtt.MQTTv311)
+    mqttc.username_pw_set(username, password=password)
+    mqttc.connect(SHIFTR_BROKER, port=port, keepalive=keepalive)
```

Comment:
I think that need inside a `yield from hass.async_add_job()` to not block the loop.